process.cpp

```diff
@@ -350,3 +350,29 @@ void procclean() {
   // delete conninode;
   prg_cache_clear();
 }
+
+void remove_timed_out_processes() {
```

Comment:
I like how factoring out this to process.cpp makes the loop in cui.cpp simpler, but it does mean we're traversing the 'processes' list twice. I suspect this will have no real impact on performance (due to cpu caches etc), but it'd be nice to be able to measure this somehow :/